app/resources/views.py

```diff
@@ -45,18 +45,81 @@ def add():
                             user_id=int(current_user.get_id()))
         db.session.add(resource)
         db.session.commit()
-        return redirect(url_for('resources.index'))
-    return render_template('resources/add_resource.html', form=form)
+        return redirect(url_for('resources.read_resource',
+                                resource_id=resource.id))
+    return render_template('resources/create_resource.html', form=form)
 
 
-@resources.route('/resource/<int:resource_id>')
+@resources.route('/read/<int:resource_id>')
 @login_required
-def show_resource(resource_id):
-    """
-    Show the resource with the given id, the id is an integer.
-    """
+def read_resource(resource_id):
     resource = Resource.query.get_or_404(resource_id)
-    address = Address.query.get(resource.address_id)
-    user = User.query.get(resource.user_id)
-    return render_template('resources/view_resource.html', resource=resource,
-                           address=address, user=user)
+    address = resource.address
+    user = resource.user
+    # TODO: base template for reviews.
+    return render_template('resources/read_resource.html', resource=resource,
+                           address=address, user=user,
+                           current_user_id=int(current_user.get_id()))
+
+
+@resources.route('/review/create/<int:resource_id>', methods=['GET', 'POST'])
+@login_required
+def create_review(resource_id):
+    resource = Resource.query.get_or_404(resource_id)
+    address = resource.address
+    user = resource.user
+    form = ReviewForm()
+    if form.validate_on_submit():
+        review = ResourceReview(timestamp=datetime.now(),
+                                content=form.content.data,
+                                rating=form.rating.data,
+                                resource_id=resource.id,
+                                user_id=int(current_user.get_id()))
+        db.session.add(review)
+        db.session.commit()
+        return redirect(url_for('resources.read_resource',
+                                resource_id=resource.id))
+    return render_template('resources/create_review.html', resource=resource,
+                           address=address, user=user, form=form)
+
+
+@resources.route('/review/update/<int:review_id>', methods=['GET', 'POST'])
+@login_required
+def update_review(review_id):
+    review = ResourceReview.query.get_or_404(review_id)
+    resource = review.resource
+    if int(current_user.id) != review.user.id:
```

Comment:
ru sure? looks like it returns unicode

def get_id(self):
        try:
            return unicode(self.id)  # python 2